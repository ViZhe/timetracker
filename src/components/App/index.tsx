
import React, { Component } from 'react';
import { Layout } from 'antd';

import SiderMenu from '../../components/SiderMenu';
import GlobalHeader from '../../components/GlobalHeader';


type Props = {
};

type State = {
  collapsed: boolean;
};

class App extends Component<Props, State> {
  readonly state: State = {
    collapsed: false,
  };

  toggle = (): void => {
    this.setState({
      collapsed: !this.state.collapsed,
    });
  }

  render() {
    const { children } = this.props;

    return (
      <Layout>
        <SiderMenu collapsed={this.state.collapsed} onCollapse={this.toggle} />
        <Layout>
          <GlobalHeader collapsed={this.state.collapsed} onClickTrigger={this.toggle} />
          {children}
        </Layout>
      </Layout>
    );
  }
}


export default App;
