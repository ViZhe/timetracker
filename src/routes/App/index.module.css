
.content {
  margin: 1.5rem 1rem;
  padding: 1.5rem;
  background-color: #fff;
  min-Height: 280px
}
