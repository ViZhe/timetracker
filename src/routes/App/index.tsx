
import { Empty } from 'antd';
import React, { lazy, Suspense } from 'react';
import { Route, Switch } from 'react-router-dom';

import AppLayout from '../../components/AppLayout';
import SectionLoading from '../../components/SectionLoading';


const Time = lazy(() => import('../Time'));
const Reports = lazy(() => import('../Reports'));
const Settings = lazy(() => import('../Settings'));

const App: React.FC = () => (
  <AppLayout>
    <Suspense fallback={<SectionLoading />}>
      <Switch>
        <Route path="/time" component={Time} exact={true} />
        <Route path="/reports" component={Reports} />
        <Route path="/settings" component={Settings} />
        <Route><Empty style={{ marginTop: '10rem' }} description={'Page not found'} /></Route>
      </Switch>
    </Suspense>
  </AppLayout>
);


export default App;
