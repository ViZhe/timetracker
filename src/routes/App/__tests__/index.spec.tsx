
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { Route, Switch } from 'react-router-dom';

import App from '../index';


describe('<App />', () => {
  const app: ShallowWrapper = shallow(<App />);

  it('should match snapshot', () => {
    expect(app).toMatchSnapshot();
  });

  it('should render some routes', () => {
    expect(app.find(Switch).length).toEqual(1);
    expect(app.find(Route).length).not.toBe(0);
  });

  it('should render route path time', () => {
    expect(app.find(Route).find({ path: '/time' }).length).toEqual(1);
  });

  it('should render route path reports', () => {
    expect(app.find(Route).find({ path: '/reports' }).length).toEqual(1);
  });

  it('should render route path projects', () => {
    expect(app.find(Route).find({ path: '/projects' }).length).toEqual(1);
  });

  it('should render route path settings', () => {
    expect(app.find(Route).find({ path: '/settings' }).length).toEqual(1);
  });
});
