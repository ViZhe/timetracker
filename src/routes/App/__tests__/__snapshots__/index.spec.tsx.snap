// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<AppLayout>
  <React.Suspense
    fallback={<UNDEFINED />}
  >
    <Switch>
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_ctor": [Function],
            "_result": null,
            "_status": -1,
          }
        }
        exact={true}
        path="/time"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_ctor": [Function],
            "_result": null,
            "_status": -1,
          }
        }
        path="/reports"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_ctor": [Function],
            "_result": null,
            "_status": -1,
          }
        }
        path="/projects"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_ctor": [Function],
            "_result": null,
            "_status": -1,
          }
        }
        path="/settings"
      />
      <Route>
        <Empty
          description="Page not found"
          style={
            Object {
              "marginTop": "10rem",
            }
          }
        />
      </Route>
    </Switch>
  </React.Suspense>
</AppLayout>
`;
